### PR TITLE
MuPDF font fixes

### PR DESCRIPTION
**Pull request by TnS-hun**
* fall back to FreeSans.ttf if needed
* font filenames were not freed everywhere
* font properties (bold, italics, ...) were not set

### CODE REVIEW

File: thirdparty/mupdf/external_fonts.patch [hunk 1]
```diff
@@ -379,23 +404,25 @@ index be69ce1c..a24b2122 100644
...
  			fz_throw(ctx, FZ_ERROR_GENERIC, "cannot find substitute font");
  
  		fontdesc->font = fz_new_font_from_memory(ctx, fontname, data, len, 0, 1);
 +#else
-+		char * filename = pdf_lookup_substitute_font_from_file(ctx, mono, serif, bold, italic);
++		char *filename = pdf_lookup_substitute_font_from_file(ctx, mono, serif, bold, italic);
 +		if (!filename)
 +			fz_throw(ctx, FZ_ERROR_GENERIC, "cannot find substitute font");
-+
++
```
Comment (Frenzie) [inline thirdparty/mupdf/external_fonts.patch]: I feel like a total nitpicker for pointing out this spare tab but GH kinda highlights it in the diff. >_>

Comment (TnS-hun) [reply thirdparty/mupdf/external_fonts.patch]: I should turn on trailing whitespace trimming. :)

**Comment by poire-z** [comment]
This PR introduced using FreeSans.ttf as a fallback font (and there's been no related change since).
I had a try at using FreeSerif.ttf, and it brings proper arabic or hebrew in footnote popups and HTML dict lookups, while still not breaking chinese in footnotes popups (so, I guess CJK glyphs are looked up in the other CJK font the external_fonts.patch registers).
As this PR introduced FreeSans.ttf in our MuPDF to make HTML dicts work, I guess it has no effect on PDF files, so the risk of breaking/changeing PDF reading is low.
So, should we switch from FreeSans.ttf to FreeSerif.ttf ?

**Comment by Frenzie** [comment]
Sure, why not?

**Pull request merged**